src/crate_graph.rs

```diff
@@ -94,6 +111,19 @@ impl CrateGraph {
                     .into(),
                 metadata,
             });
+            let updated_lock_content = match std::fs::exists(&lock_path)? {
```

Comment:
It's not obvious why the lock file would have changed. I assume it's a side effect of running `cargo metadata`. Can you leave a comment to explain that?